pkg/apis/serving/v1beta1/transformer_custom.go

```diff
@@ -72,6 +73,23 @@ func (c *CustomTransformer) GetContainer(metadata metav1.ObjectMeta, extensions
 	container := &c.Containers[0]
 	argumentPredictorHost := fmt.Sprintf("%s.%s", constants.DefaultPredictorServiceName(metadata.Name), metadata.Namespace)
 
+	deploymentMode, ok := metadata.Annotations[constants.DeploymentMode]
+	logger, _ := pkglogging.NewLogger("", "INFO")
+	if ok && (deploymentMode == string(constants.ModelMeshDeployment)) {
+		// Get predictor host and protocol from annotations in modelmesh deployment mode
+		argumentPredictorHost = metadata.Annotations["predictor-host"]
+		argumentPredictorProtocol := metadata.Annotations["predictor-protocol"]
```

Comment:
Should we add a `protocol` field on the `InferenceService` component spec instead of using the annotation?

Reply:
@chinhuang007 I think it would be simpler to just add an additional `predictorURL` arg to this `GetContainer` function? Then no intermediary annotations would be needed.

Reply:
@njhill GetContainer is an interface, also implemented in other places. The goal is to minimize interface change therefore use of intermediary annotations.

Reply:
@yuzisun @njhill @pvaneck The current use of intermediary annotations is aimed to minimize interface changes. Adding a new arg to GetContainer, part of ComponentImplementation interface, would require changes in 30+ files, including predictor, explainer, and a bunch of examples. Not a difficult task, I just wanted to discuss whether it is warranted before making an interface change.

If we decide a new arg should be added, I'd like to make sure the type is correct since the impact is broad. I am currently thinking InferenceServiceStatus might work?